indico/modules/rb_new/controllers/backend/blockings.py

```diff
@@ -73,9 +75,40 @@ def _process(self, timeframe, my_rooms, mine):
         return jsonify(blockings_schema.dump(blockings).data)
 
 
-class RHRoomBlocking(RHRoomBookingBase):
+class RHRoomBlockingBase(RHRoomBookingBase):
     def _process_args(self):
         self.blocking = Blocking.get_one(request.view_args['blocking_id'])
 
+
+class RHRoomBlocking(RHRoomBlockingBase):
     def _process(self):
         return jsonify(blockings_schema.dump(self.blocking, many=False).data)
+
+
+class RHBlockingAction(RHRoomBlockingBase):
+    def _process_args(self):
+        RHRoomBlockingBase._process_args(self)
+        self.action = request.view_args['action']
+        self.blocked_room = (BlockedRoom.query
+                             .with_parent(self.blocking)
+                             .filter_by(id=request.view_args['blocked_room_id'])
+                             .first_or_404())
+
+    def _process(self):
+        if self.action == 'accept':
+            self.blocked_room.approve()
+        elif self.action == 'reject':
+            self.reject()
+        return jsonify_data(flash=False, blocking=blockings_schema.dump(self.blocking, many=False).data)
```

Comment:
just use `jsonify()` here since you most likely don't need `success: true` in the response either